File: benchmark/lua/vector_dot.lua
local a = {}
local b = {}
for i = 1, 500000 do
    a[i] = i
    b[i] = i + 1
end
local sum = 0
local start = os.clock()
for i = 1, 500000 do
    sum = sum + a[i] * b[i]
end
local finish = os.clock()
print("Dot: " .. sum)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/perl/vector_dot.pl
use Time::HiRes qw(time);
my (@a, @b);
for my $i (0..499999) { $a[$i] = $i + 1; $b[$i] = $i + 2; }
my $start = time();
my $s = 0;
for my $i (0..499999) { $s += $a[$i] * $b[$i]; }
my $end = time();
my $ms = ($end - $start) * 1000;
print "Dot: $s
";
printf "Time: %.0fms
", $ms;
